product_to_bizerba_scale/models/product_scale_log.py

```diff
@@ -0,0 +1,347 @@
+# Copyright (C) 2014 GRAP (http://www.grap.coop)
+# © 2017 Coop IT Easy (http://www.coopiteasy.be)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# @author: Houssine BAKKALI (https://github.com/houssine78)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import codecs
+import os
+import logging
+from datetime import datetime
+
+from odoo import api, fields, models, tools, _
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from ftplib import FTP
+except ImportError:
+    _logger.warning(
+        "Cannot import 'ftplib' Python Library. 'product_to_bizerba_scale' module will not work properly.")
+
+
+class ProductScaleLog(models.Model):
+    _name = 'product.scale.log'
+    _inherit = 'mail.activity.mixin'
+    _order = 'log_date desc, id desc'
+
+    _EXTERNAL_SIZE_ID_RIGHT = 4
+
+    _DELIMITER = '#'
+
+    _ACTION_SELECTION = [
+        ('create', 'Creation'),
+        ('write', 'Update'),
+        ('unlink', 'Deletion'),
+    ]
+
+    _ACTION_MAPPING = {
+        'create': 'C',
+        'write': 'C',
+        'unlink': 'S',
+    }
+
+    _ENCODING_MAPPING = {
+        'iso-8859-1': '\r\n',
+        'cp1252': '\n',
+        'utf-8': '\n',
+    }
+
+    _EXTERNAL_TEXT_ACTION_CODE = 'C'
+
+    _EXTERNAL_TEXT_DELIMITER = '#'
+
+    log_date = fields.Datetime(
+        string='Log Date',
+        required=True
+    )
+    scale_system_id = fields.Many2one(
+        comodel_name='product.scale.system',
+        string='Scale System',
+        required=True
+    )
+    send_product_image = fields.Boolean(
+        string='Send product image'
+    )
+    product_id = fields.Many2one(
+        'product.template',
+        string='Product'
+    )
+    product_text = fields.Text(
+        compute='_compute_text',
+        string='Product Text',
+        store=True,
+    )
+    external_text = fields.Text(
+        compute='_compute_text',
+        string='External Text',
+        store=True,
+    )
+    external_text_display = fields.Text(
+        compute='_compute_text',
+        string='External Text (Display)',
+        store=True,
+    )
+    action = fields.Selection(
+        selection=_ACTION_SELECTION,
+        string='Action',
+        required=True
+    )
+    sent = fields.Boolean(
+        string='Is Sent'
+    )
+    last_send_date = fields.Datetime(
+        string='Last Send Date'
+    )
+
+    @api.noguess
+    def _auto_init(self):
+        # FIXME on install -> psycopg2.ProgrammingError: relation "product_scale_log" does not exist
```

Comment:
didn't found a fix for this ?

Reply:
ouh ! I completely forgot about this one. Thank you for the reminder! I'll try to find a fix and ping you once it's done

Reply:
I've looked into it.
- `DELETE FROM product_scale_log` is an `Unsafe query: 'Delete' statement without 'where' clears all data in the table`
- we can use instead `DROP TABLE IF EXISTS product_scale_log`: it doesn't raise `psycopg2.ProgrammingError`

But, do we want this behavior ? If we execute an `update-module product_to_bizerba_scale`, all logs will be deleted, including the ones that might not have been sent to the FTP

Reply:
I remember, that there is a problem with the `product.scale.log` table when performing an update on this module. So it blocks an update all.

I'm curious to see if this issue is still present after your refactoring. Maybe we should give it a try. Then bring back this hack if the problem still persist.